inst/tests/tests.Rraw

```diff
@@ -16770,6 +16770,8 @@ test(2132.2, fifelse(TRUE, 1, s2),       error = "S4 class objects (except nanot
 test(2132.3, fcase(TRUE, s1, FALSE, s2), error = "S4 class objects (except nanotime) are not supported. Please see https://github.com/Rdatatable/data.table/issues/4131.")
 rm(s1, s2, class2132)
 
+# setlevels, #2219
+test(2133.1, setlevels(as.factor(c("A", "A", "B", "B", "B", "C")), c("X", "Y", "Z")), as.factor(c("X", "X", "Y", "Y", "Y", "Z")))
```

Comment:
The issue was written specifically about how `setlevels` improves vs `setattr` because of the consistency checks; can we add some tests of these here?

Reply:
Now that I look in more details at `setattrib` and `setlevels` I am a bit confused by these functions. Maybe you can shed some light on the following code :

```R
x = as.factor(c("A","A","B","B","B","C"))
print(setlevels(x, c("X","X","Y","Z")))
# X X X X X Y
# Levels: X Y Z
```
That is not correct, is it? `setattr` behaves the same as it is built on top of `setlevels`.

Do I need to implement some checks at C level? I am happy to do it but I am bit confused now. I think I went a bit too fast!

Reply:
Indeed that looks wrong!

I think my understanding of the motivation for `setlevels` was to error for that example, since `X` is duplicated. If not an error, `unique(value)` should be used.

Reply:
Well that is not a problem, I can make it error ☺. Anything else?

Reply:
Just to be clear, that is what I have with `setattr` as well:
```R
print(setattr(x, "levels", c("X","X","Y","Z")))
# [1] X X X X X Y
# Levels: X Y Z
```
I can correct everything if you tell me exactly what each function is suppose to do. Thanks

Reply:
Made some changes,  let me know if anything else is needed. Thanks.